### PR TITLE
Initial basic functionality of short form

### PR DESCRIPTION
**Pull request by rraub**
This should be the bare minimum to get the basic "ask a question" short form working. Nothing glamorous yet, just enough to get shipped and we can work on improving it as we go.

This plugin adds a shortcode `[asu-rfi-form]` and a settings page for defining the `source_id` for the side.

Todo:
- [x] client side validation of fields
- [x] cache control settings
- [x] site wide settings for source_id

screen shot 2016-12-12 at 3 38 20 pm

Success Banner:
screen shot 2016-12-12 at 3 38 35 pm

Error Banner:
screen shot 2016-12-12 at 3 42 14 pm

Settings Menu Location:
screen shot 2016-12-13 at 12 43 24 pm

### CODE REVIEW

**Comment by rraub** [comment]
@nathan-rollins I dont quite like the wording of the question "I am going to be a.." but having some optional section for type of student is nice. Do you have any ideas for a better phrase?

File: src/services/asu-degree-service.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+   * enrollment terms
+   * These are the peoplesoft term identifiers
+   *  "2101" for Spring, 2010, or  3
+   *  "2104" for Summer, 2010, or  3
+   *  "2107" for Fall, 2010, or  2
+   *  "2109" for Winter, 2010
+   *  "2177" for Summer, 2017
+   */
+  public static function get_available_enrollment_terms() {
+    $semseter_names = array(
```
Comment (nathan-rollins) [inline src/services/asu-degree-service.php]: FYI: spelling typo

File: src/views/rfi-form/form.handlebars [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        </div>
+      </div>
+      <div class="form-group has-feedback required">
+        <label class="control-label col-sm-3" for="question">Additional Questions?</label>
+        <div class="col-sm-9">
+          <textarea type="text" class="form-control" id="question" name="question" maxlength="500"></textarea>
+        </div>
+      </div>
+      <div class="form-group has-feedback">
+        <label class="control-label col-sm-3" for="submit"></label>
```
Comment (nathan-rollins) [inline src/views/rfi-form/form.handlebars]: Is this label useful for accessibility/screen readers? Maybe an alternate aria attribute could be leveraged to make the submit button obvious (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/Basic_form_hints).

File: src/views/rfi-form/simple-request-info-form.handlebars [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            {{#each student_types }}
+              <option value="{{ value }}">{{ label }}</option>
+            {{/each}}
+          </select>
+        </div>
+      </div>
+      
+      
+      <div class="form-group has-feedback">
+        <label class="control-label col-sm-3" for="submit"></label>
```
Comment (nathan-rollins) [inline src/views/rfi-form/simple-request-info-form.handlebars]: Same possible issue with accessibility for this label.

**Comment by rraub** [comment]
I've replaced the empty label for the submit button with an aria-label; it is kinda redundant but its more explicit for some older screen readers. (Much better than an empty label) Also I've added `aria-required="true"` for screen readers that don't yet support html 5.

**Pull request merged**